lib/auth/clt.go

```diff
@@ -2670,6 +2650,73 @@ func (c *Client) Ping(ctx context.Context) (proto.PingResponse, error) {
 	return *rsp, nil
 }
 
+// ClientServices implements IClientServices interface.
+type ClientServices struct {
+	roundtrip.Client
+}
+
+// PostJSON is a generic method that issues http POST request to the server
+func (c *ClientServices) PostJSON(
+	endpoint string, val interface{}) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PostJSON(context.TODO(), endpoint, val))
+}
+
+// PutJSON is a generic method that issues http PUT request to the server
+func (c *ClientServices) PutJSON(
+	endpoint string, val interface{}) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PutJSON(context.TODO(), endpoint, val))
+}
+
+// PostForm is a generic method that issues http POST request to the server
+func (c *ClientServices) PostForm(
+	endpoint string,
+	vals url.Values,
+	files ...roundtrip.File) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.PostForm(context.TODO(), endpoint, vals, files...))
+}
+
+// Get issues http GET request to the server
+func (c *ClientServices) Get(u string, params url.Values) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.Get(context.TODO(), u, params))
+}
+
+// Delete issues http Delete Request to the server
+func (c *ClientServices) Delete(u string) (*roundtrip.Response, error) {
+	return httplib.ConvertResponse(c.Client.Delete(context.TODO(), u))
+}
+
+// Endpoint returns a URL constructed from parts and version appended, e.g.
+func (c *ClientServices) Endpoint(params ...string) string {
+	return c.Client.Endpoint(params...)
+}
+
+// PostSessionSlice allows clients to submit session stream chunks to the audit log
+// (part of evets.IAuditLog interface)
+//
+// The data is POSTed to HTTP server as a simple binary body (no encodings of any
+// kind are needed)
+func (c *ClientServices) PostSessionSlice(slice events.SessionSlice) error {
```

Comment:
this method seems out-of-place here.
all others are generic while this is type-specific.

how about a new method `PostBytes` and another method `func (c *Client) PostSessionSlice` that calls into this?